src/intrinsics/prepack/global.js

```diff
@@ -263,22 +263,12 @@ export default function(realm: Realm): void {
         "global.__evaluatePureFunction",
         "__evaluatePureFunction",
         0,
-        (context, [functionValue, callback]) => {
+        (context, [functionValue]) => {
+          invariant(!realm.isInPureScope(), "__evaluatePureFunction cannot be nested in another pure scope");
```

Comment:
This sounds like a user facing error message, so putting it in an invariant is wildly wrong.